amazon-q/main.tf

```diff
@@ -0,0 +1,294 @@
+  type        = bool
+  description = "Whether to use screen for running Amazon Q in the background."
+  default     = false
+}
+
+variable "experiment_use_tmux" {
+  type        = bool
+  description = "Whether to use tmux instead of screen for running Amazon Q in the background."
+  default     = false
+}
+
+variable "experiment_report_tasks" {
+  type        = bool
+  description = "Whether to enable task reporting."
+  default     = false
+}
+
+variable "experiment_pre_install_script" {
+  type        = string
+  description = "Custom script to run before installing Amazon Q."
+  default     = null
+}
+
+variable "experiment_post_install_script" {
+  type        = string
+  description = "Custom script to run after installing Amazon Q."
+  default     = null
+}
+
+variable "experiment_auth_tarball" {
+  type        = string
+  description = "Base64 encoded, zstd compressed tarball of a pre-authenticated ~/.local/share/amazon-q directory. After running `q login` on another machine, you may generate it with: `cd ~/.local/share/amazon-q && tar -c . | zstd | base64 -w 0`"
+  default     = "tarball"
+}
+
+locals {
+  encoded_pre_install_script  = var.experiment_pre_install_script != null ? base64encode(var.experiment_pre_install_script) : ""
+  encoded_post_install_script = var.experiment_post_install_script != null ? base64encode(var.experiment_post_install_script) : ""
+  # We need to use allowed tools to limit the context Amazon Q receives.
+  # Amazon Q can't handle big contexts, and the `create_template_version` tool
+  # has a description that's too long.
+  mcp_json         = <<EOT
+    {
+      "mcpServers": {
+        "coder": {
+          "command": "coder",
+          "args": ["exp", "mcp", "server", "--allowed-tools", "coder_report_task"],
+          "env": {
+            "CODER_MCP_APP_STATUS_SLUG": "amazon-q"
+          }
+        }
+      }
+    }
+  EOT
+  encoded_mcp_json = base64encode(local.mcp_json)
+}
+
+
+
+# Install and Initialize Amazon Q
+resource "coder_script" "amazon_q" {
+  agent_id     = var.agent_id
+  display_name = "Amazon Q"
+  icon         = var.icon
+  script       = <<-EOT
+    #!/bin/bash
+    set -o errexit
+    set -o pipefail
+
+    # Function to check if a command exists
+    command_exists() {
+      command -v "$1" >/dev/null 2>&1
+    }
+
+    # Run pre-install script if provided
+    if [ -n "${local.encoded_pre_install_script}" ]; then
+      echo "Running pre-install script..."
+      echo "${local.encoded_pre_install_script}" | base64 -d > /tmp/pre_install.sh
+      chmod +x /tmp/pre_install.sh
+      /tmp/pre_install.sh
+    fi
+
+    # Install Amazon Q if enabled
+    if [ "${var.install_amazon_q}" = "true" ]; then
+      echo "Installing Amazon Q..."
+      PREV_DIR="$PWD"
+      TMP_DIR="$(mktemp -d)"
+      cd "$TMP_DIR"
+      curl --proto '=https' --tlsv1.2 -sSf "https://desktop-release.q.us-east-1.amazonaws.com/${var.amazon_q_version}/q-x86_64-linux.zip" -o "q.zip"
```

Comment:
We should probably do an architecture check here since we only support x64 Linux

Reply:
If they have an arm64 asset, let's do that too. I have seen recent PRs of users adding arm64 support to other modules.